src/sys/socket/mod.rs

```diff
@@ -846,6 +851,13 @@ pub struct Timestamps {
     pub hw_raw: TimeSpec,
 }
 
+// Defined in `linux/tls.h`
+#[cfg(all(target_os = "linux"))]
+const TLS_GET_RECORD_TYPE: c_int = 2;
```

Comment:
This definition should be pushed into the libc crate first, and then you should update Nix to use the new libc verison.  It isn't Nix's place to define C APIs.  Also, do you know if this same interface is defined on Android?  If so, include that in the `#[cfg]` directive.

Reply:
done!